src/plugins/intel_npu/src/plugin/npuw/weights_bank.cpp

```diff
@@ -74,77 +71,113 @@ ov::Tensor Bank::get(int64_t uid, const std::string& device) {
+            evaluate_cpu(device_bank, to_process);
+        } else {
+            evaluate_and_allocate_on_device(device_bank, to_process, device_for_alloc);
+        }
+    }
+}
+
+void Bank::evaluate_cpu(Bank::DeviceBank& device_bank, const std::vector<LazyTensor>& to_process) {
+    std::unique_lock guard(m_mutex);
+
+    ov::parallel_for(to_process.size(), [&](std::size_t idx) {
+        const auto& lt = to_process[idx];
+        auto iter_device_registered = device_bank.registered_tensors.find(lt);
+        NPUW_ASSERT(iter_device_registered != device_bank.registered_tensors.end() &&
+                    "Tensor should be registered first!");
+        auto uid = iter_device_registered->second;
+        device_bank.storage.at(uid).tensor = lt.eval();
+        const_cast<LazyTensor&>(lt).detach();
+    });
+}
+
+void Bank::evaluate_and_allocate_on_device(Bank::DeviceBank& device_bank,
+                                           const std::vector<LazyTensor>& to_process,
+                                           const std::string& device) {
+    std::unique_lock guard(m_mutex);
+
+    std::vector<TensorToAllocate> uids_to_allocated(uid_count);
+
+    // Allocation needed
+    for (std::size_t idx = 0; idx < to_process.size(); ++idx) {
+        const auto& lt = to_process[idx];
+        auto iter_device_registered = device_bank.registered_tensors.find(lt);
+        NPUW_ASSERT(iter_device_registered != device_bank.registered_tensors.end() &&
+                    "Tensor should be registered first!");
+        auto uid = iter_device_registered->second;
+        uids_to_allocated[uid] = {lt.eval_meta(), ov::Tensor(), true};
+    }
 
-            auto remote_ctx = m_core->get_default_context(device_for_alloc)._ptr;
-            remote_tensor =
-                remote_ctx->create_host_tensor(transformed_tensor.get_element_type(), transformed_tensor.get_shape());
-            allocated_tensor = ov::make_tensor(remote_tensor);
-            device_bank.storage[device_bank.registered_tensors.at(lt)].tensor = allocated_tensor;
-            dev_guard.unlock();  // Unlock the guard, map update is done - copy can continue in parallel
+    // Allocate memory sequentially - in order of UID
+    auto remote_ctx = m_core->get_default_context(device)._ptr;
+    for (std::size_t uid = 0; uid < uids_to_allocated.size(); ++uid) {
+        auto& allocated = uids_to_allocated[uid];
+        if (allocated.should_be_processed) {
+            ov::SoPtr<ov::ITensor> remote_tensor =
+                remote_ctx->create_host_tensor(allocated.meta.type, allocated.meta.shape);
+            uids_to_allocated.at(uid) = {allocated.meta, ov::make_tensor(remote_tensor), true};
+        }
+    }
 
-            transformed_tensor.copy_to(allocated_tensor);
+    // Evaluate and copy into the device memory
+    ov::parallel_for(uids_to_allocated.size(), [&](std::size_t uid) {
+        auto& allocated = uids_to_allocated[uid];
+        if (allocated.should_be_processed) {
```

Comment:
when it is `false`, again?

Reply:
https://github.com/openvinotoolkit/openvino/pull/29039/files#diff-3f12fee0c24eaba5d803fa98acb3d04f9b8f4a70d2b1522ce23d3d6f86e6e96eR135

Reply:
Reworked again. Added uid field to TensorToAllocate + sorting based on uid